network/anomaly_detector_model.py

```diff
@@ -40,10 +40,10 @@ def custom_objective(y_pred, y_true):
     anomal_segments_scores = y_pred[anomal_vids_indices].squeeze(-1)  # (batch/2, 32, 1)
 
     # get the max score for each video
-    normal_segments_scores_maxes = normal_segments_scores.max(dim=-1)[0]
-    anomal_segments_scores_maxes = anomal_segments_scores.max(dim=-1)[0]
+    normal_segments_scores_maxes = normal_segments_scores.topk(k=2, dim=-1)[0] + normal_segments_scores.topk(k=2, largest=False, dim=-1)[0]
+    anomal_segments_scores_maxes = anomal_segments_scores.topk(k=2, dim=-1)[0] + normal_segments_scores.topk(k=2, largest=False, dim=-1)[0]
 
-    hinge_loss = 1 - anomal_segments_scores_maxes + normal_segments_scores_maxes
+    hinge_loss = 4 - torch.sum(anomal_segments_scores_maxes, dim=-1) + torch.sum(normal_segments_scores_maxes, dim=-1)
```

Comment:
This is not the loss function described in the original paper

Reply:
I tried to make several variations from the loss function in the original paper. It's one of them.
I fixed it with the loss function from the original paper in the latest commit!